File: ios-native/Jottery/Jottery/Models/AttachmentRef.swift
import Foundation

/// Attachment reference (metadata only — data lives in blob store or sync payload).
struct AttachmentRef: Codable, Identifiable, Equatable, Hashable {
    let id: String
    var filename: String       // Encrypted when stored, decrypted in memory
    var mimeType: String
    var size: Int
    var data: String           // Reference ID to encrypted blob store
    var thumbnailData: String? // Optional thumbnail reference
}

File: android-native/app/src/main/java/com/jottery/android/model/AttachmentRef.kt
package com.jottery.android.model

import kotlinx.serialization.Serializable

/**
 * Reference to an encrypted attachment blob. Stored as JSON within notes/versions.
 * Not a Room entity — embedded in notes.attachments JSON column.
 */
@Serializable
data class AttachmentRef(
    val id: String,
    var filename: String,
    val mimeType: String,
    val size: Int,
    val data: String,
    val thumbnailData: String? = null,
)
